app/validators/blocks/block_validator.py

```diff
@@ -62,3 +68,37 @@ def validate_answer_source_reference(self, identifiers):
                 == self.block["id"]
             ):
                 self.add_error(self.ANSWER_SELF_REFERENCE, referenced_id=identifier)
+
+    def validate_redirect_to_list_add_block_params(self):
+        collector_questions = self.questionnaire_schema.get_all_questions_for_block(
+            self.block
+        )
+
+        for collector_question in collector_questions:
+            for collector_answer in collector_question["answers"]:
+                for option in collector_answer.get("options", []):
+                    action = option.get("action")
+
+                    if action:
+                        params = action.get("params")
+                        if not params:
+                            self.add_error(
+                                self.ACTION_PARAMS_MISSING, block_id=self.block
+                            )
+                            continue
+
+                        list_name = params.get("list_name")
+
+                        if (
+                            list_name
+                            and list_name not in self.questionnaire_schema.list_names
+                        ):
+                            self.add_error(self.LIST_NAME_MISSING, list_name=list_name)
+
+                        block_id = params.get("block_id")
+
+                        if (
+                            block_id
+                            and block_id not in self.questionnaire_schema.block_ids
+                        ):
+                            self.add_error(self.BLOCK_ID_MISSING, block_id=block_id)
```

Comment:
Once the comment about updating the JSON schema is done, this can be simplfied to:
```py
if list_name := params["list_name"] not in self.questionnaire_schema.list_names:
    self.add_error(self.LIST_NAME_MISSING, list_name=list_name)

if block_id := params["block_id"] not in self.questionnaire_schema.block_ids:
    self.add_error(self.BLOCK_ID_MISSING, block_id=block_id)
```
However, the walrus operator means you will need to update black formatting to latest version so, you could do `list_name = params["list_name"]` for now if you wanted to keep updating the package separate.